server.js

```diff
@@ -1,33 +1,144 @@
+  const { title, cast, country, director } = req.query
+  let shows = netflixData
+
+  // Filters so the user can search for items in the endpoint
+  if (title) {
+    shows = shows.filter(show => show.title.toString().includes(title))
+  }
+  if (cast) {
+    shows = shows.filter(show => show.cast.toString().includes(cast))
+  }
+  if (country) {
+    shows = shows.filter(show => show.country.toString().includes(country))
+  }
+  if (director) {
+    shows = shows.filter(show => show.director.toString().includes(director))
+  }
+
+  // Show an error message if the returning array is empty (no search result)
+  if (shows.length === 0) {
+    res.status(404).send('Not found')
+
+  // Show all data
+  } else {
+    res.json(shows)
+  }
+})
+
+// ** Endpoint to get only movies **
+
+app.get('/movies', (req, res) => {
+  let movies = netflixData.filter(show => show.type === "Movie")
+  const { title, cast, country, director } = req.query
+
+  // Filters so the user can search for items in the movies endpoint
+  if (title) {
+    movies = movies.filter(show => show.title.toString().includes(title))
+  }
+  if (cast) {
+    movies = movies.filter(show => show.cast.toString().includes(cast))
+  }
+  if (country) {
+    movies = movies.filter(show => show.country.toString().includes(country))
+  }
+  if (director) {
+    movies = movies.filter(show => show.director.toString().includes(director))
+  }
+
+  // Show an error message if the returning array is empty (no search result)
+  if (movies.length === 0) {
+    res.status(404).send('Not found')
+
+  // Show all data for the movies
+  } else {
+    res.json(movies)
+  }
+})
+
+// ** Endpoint to get only TV-shows **
+
+app.get('/tv-shows', (req, res) => {
+  let tvShows = netflixData.filter(show => show.type === "TV Show")
+  const { title, cast, country, director } = req.query
+
+  // Filters so the user can search for items in the tv-shows endpoint
+  if (title) {
+    tvShows = tvShows.filter(show => show.title.toString().includes(title))
+  }
+  if (cast) {
+    tvShows = tvShows.filter(show => show.cast.toString().includes(cast))
+  }
+  if (country) {
+    tvShows = tvShows.filter(show => show.country.toString().includes(country))
+  }
+  if (director) {
+    tvShows = tvShows.filter(show => show.director.toString().includes(director))
+  }
+
+  // Show an error message if the returning array is empty (no search result)
+  if (tvShows.length === 0) {
+    res.status(404).send('Not found')
+
+  // Show all data for the movies
+  } else {
+    res.json(tvShows)
+  }
 })
 
-// Start the server
+// ** Endpoint to get only one show by title **
+
+app.get('/shows/title/:title', (req, res) => {
+  const { title } = req.params
+  const showTitle = netflixData.find(show => show.title === title)
+
+  if (!showTitle) {
+    res.status(404).send(`No show with the title ${title} exists`)
+  }
+  res.json(showTitle)
+})
+
+// ** Endpoint to get only one show by id **
+
+app.get('/shows/id/:id', (req, res) => {
```

Comment:
when you use a primary key, like id, you should not specify the key in the address, so it should be `/shows/:id`